### ISSUE TITLE
Sync randomSpread bullets between client and server

### ISSUE THREAD

**Issue by SnailPerson** [DonBruce64/MinecraftTransportSimulator#1462]
<img width="830" alt="fffew" src="https://user-images.githubusercontent.com/97041048/208553244-c1e09123-6e8f-4bd0-a158-44ef66f9a88b.png">

**Comment by boot2big** [comment]
Actually I'd like to point out that this bug is a bit deeper than it seems... bullets in general seem to be slightly desynced between the client and server, for singleplayer.

With dedicated server testing it had seemed this wasn't the case however, so what's truly causing it is beyond me. It almost seems like sometimes bullets do account for player velocity, and sometimes they don't.

One other annoying side effect I've noticed from this as a result of a new weapon I've added is that sometimes, bullets on the client will think they're tracking an entity, while on the server they've simply been free-fired. It's sickening to watch an advanced homing missile make direct contact with a slime, and watch it do... ABSOLUTELY NOTHING to the slime while a random village in the background has a random house just, explode!

TL;DR This may be deeper rooted beyond _just_ bulletspread, I think. I've just been neglecting to ticket it because it seemed it'd be hard to work through this, and when I had first discovered it there was... major discord drama. So I figured it was a bit of a bad time.

**Comment by DonBruce64** [comment]
I know why the homing issue happens: that's because there's slight differences in player position and entity position on the server and client.  So you might have lock-on on the client, but not so on the server.  To fix that, I'd need to send a packet with the lock-on state to the client to have it show the true state.... but that's a different issue for a different ticket :sans:.

**Comment by DonBruce64** [comment]
Fixed in V22.6.0.
